### ISSUE TITLE
Add WASM support

### ISSUE THREAD

**Issue by StefanOltmann** [goncalossilva/kotlinx-resources#91]
Please add support for WASM & WASI targets

**Comment by lppedd** [comment]
Vouch for this. We'd like to add `wasmJs` and `wasmWasi` as test targets.

**Comment by goncalossilva** [comment]
I'll look into it soon. 👍

**Comment by goncalossilva** [comment]
Blocked on [this](https://youtrack.jetbrains.com/issue/KT-63014/Running-tests-with-wasmJs-in-1.9.20-requires-Chrome-Canary#focus=Comments-27-8659998.0-0), but I'm hoping there is a straightforward workaround (e.g., using a Canary version of Node). I've asked on Kotlin's Slack as well.

**Comment by lppedd** [comment]
@goncalossilva using Canary is what official Kotlin libs are doing, yup!

**Comment by goncalossilva** [comment]
Yeah, that's the way to move forward. I've rewritten `ResourceNode` but it appears I'll also have to rewrite `ResourceBrowser` if I want it shared between JS and Wasm.

**Comment by lppedd** [comment]
Left a couple comments in the new commits.

**Comment by iTanChi** [comment]
@goncalossilva please update to support this feature. I really need this to enable wasm targets on our repos

**Comment by goncalossilva** [comment]
@iTanChi I'm afraid I've been travelling for work and unavailable to continue working on this, but I plan to pick it up again later this month, before the upcoming KotlinConf.

**Comment by mipastgt** [comment]
Any news on this front? I'd like to see support for wasmJs to get some of my tests working.

**Comment by goncalossilva** [comment]
No progress on my end, I'm afraid. I haven't been able to prioritize this, as I'm not actively using WASM on any personal or work project. I still want to get around to it eventually, but of course, contributions would be most welcome.

**Comment by mipastgt** [comment]
I am also not yet really using WASM too but I am trying to pick up the necessary pieces and see where we are :-)

**Comment by lppedd** [comment]
Thank you @goncalossilva!

**Comment by goncalossilva** [comment]
My pleasure, folks. 💪 

I'm just working on a couple of additions and stabilizing the codebase and I'll do a release with this soon.
